diff --git a/src/gb/audio.c b/src/gb/audio.c
index 6a97d8bea..f9165e896 100644
--- a/src/gb/audio.c
+++ b/src/gb/audio.c
@@ -1118,8 +1118,6 @@ void GBAudioPSGDeserialize(struct GBAudio* audio, const struct GBSerializedPSGSt
 			audio->ch4.lastEvent = currentTime + (when & (cycles - 1)) - cycles;
 		}
 	}
-	audio->ch4.nSamples = 0;
-	audio->ch4.samples = 0;
 }
 
 void GBAudioSerialize(const struct GBAudio* audio, struct GBSerializedState* state) {
diff --git a/src/gba/audio.c b/src/gba/audio.c
index 1e913554f..271ebb30c 100644
--- a/src/gba/audio.c
+++ b/src/gba/audio.c
@@ -495,16 +495,6 @@ void GBAAudioSerialize(const struct GBAAudio* audio, struct GBASerializedState*
 void GBAAudioDeserialize(struct GBAAudio* audio, const struct GBASerializedState* state) {
 	GBAudioPSGDeserialize(&audio->psg, &state->audio.psg, &state->audio.flags);
 
-	uint16_t reg;
-	LOAD_16(reg, GBA_REG_SOUND1CNT_X, state->io);
-	GBAIOWrite(audio->p, GBA_REG_SOUND1CNT_X, reg & 0x7FFF);
-	LOAD_16(reg, GBA_REG_SOUND2CNT_HI, state->io);
-	GBAIOWrite(audio->p, GBA_REG_SOUND2CNT_HI, reg & 0x7FFF);
-	LOAD_16(reg, GBA_REG_SOUND3CNT_X, state->io);
-	GBAIOWrite(audio->p, GBA_REG_SOUND3CNT_X, reg & 0x7FFF);
-	LOAD_16(reg, GBA_REG_SOUND4CNT_HI, state->io);
-	GBAIOWrite(audio->p, GBA_REG_SOUND4CNT_HI, reg & 0x7FFF);
-
 	LOAD_32(audio->chA.internalSample, 0, &state->audio.internalA);
 	LOAD_32(audio->chB.internalSample, 0, &state->audio.internalB);
 	memcpy(audio->chA.samples, state->samples.chA, sizeof(audio->chA.samples));
diff --git a/src/gba/io.c b/src/gba/io.c
index 658036e4c..86286864a 100644
--- a/src/gba/io.c
+++ b/src/gba/io.c
@@ -244,8 +244,8 @@ static const int _isWSpecialRegister[GBA_REG(INTERNAL_MAX)] = {
 	/* 04 */ 0, 0, 0, 0, 0, 0, 0, 0,
 	/* 05 */ 0, 0, 0, 0, 0, 0, 0, 0,
 	/*    Audio */
-	/* 06 */ 0, 0, 1, 0, 0, 0, 1, 0,
-	/* 07 */ 0, 0, 1, 0, 0, 0, 1, 0,
+	/* 06 */ 1, 1, 1, 0, 1, 0, 1, 0,
+	/* 07 */ 1, 0, 1, 0, 1, 0, 1, 0,
 	/* 08 */ 0, 0, 1, 0, 0, 0, 0, 0,
 	/* 09 */ 1, 1, 1, 1, 1, 1, 1, 1,
 	/* 0A */ 1, 1, 1, 1, 0, 0, 0, 0,
